allauth/socialaccount/providers/metamask/views.py

```diff
@@ -0,0 +1,71 @@
+import json
+from django.http import JsonResponse
+import random
+import string
+from django.utils.translation import ugettext_lazy as _
+from django.views.decorators.csrf import csrf_exempt
+
+from django.core.exceptions import ImproperlyConfigured
+from django.contrib.auth import logout
+
+from allauth.socialaccount import app_settings, providers
+from allauth.socialaccount.helpers import (
+    complete_social_login,
+    render_authentication_error,
+)
+from allauth.socialaccount.models import SocialLogin, SocialToken
+from .provider import MetamaskProvider
+from django.views.decorators.http import require_http_methods
+
+# web3 declarations
+from web3 import Web3
+from eth_account.messages import encode_defunct
+from django.utils.translation import ugettext_lazy as _
+
+@csrf_exempt
+@require_http_methods(["GET","POST"])
+def login_api(request):
+    extra_data = request.body.decode('utf-8')
+    data = json.loads(extra_data)
+    request.uid = data["account"]
+    request.process = data["process"]
```

Comment:
The code above will start crashing if the data passed is not JSON, or if keys are missing. Better properly validate the input and handle accordingly.

Reply:
Error handling added with latest PR to my local branch